sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/models/AgentPoolData.java

```diff
@@ -0,0 +1,497 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.containerservice.models;
+
+import com.azure.resourcemanager.containerservice.fluent.models.AgentPoolInner;
+import com.azure.resourcemanager.resources.fluentcore.arm.ResourceUtils;
+import com.azure.resourcemanager.resources.fluentcore.model.HasInnerModel;
+import com.azure.resourcemanager.resources.fluentcore.utils.ResourceManagerUtils;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+/**
+ * The client-side data of an agent pool.
+ */
+public class AgentPoolData implements AgentPool, HasInnerModel<AgentPoolInner> {
+
+    private final AgentPoolInner innerModel;
+
+    /**
+     * Creates an instance of agent pool data.
+     */
+    public AgentPoolData() {
+        this(new AgentPoolInner());
+    }
+
+    /**
+     * Creates an instance of agent pool data.
+     *
+     * @param innerModel the inner model of agent pool.
+     */
+    public AgentPoolData(AgentPoolInner innerModel) {
```

Comment:
Not sure if we want to expose this constructor to user. Tried some alternatives but failed checkstyle.

Reply:
You may try to have another `AgentPoolDataImpl` in implementation package (inherit this one), and in KubernetesClusterImpl.java, init the `AgentPoolDataImpl` with this additional ctor.